app/validators/routing/types.py

```diff
@@ -43,13 +43,22 @@
 }
 
 
-def resolve_value_source_json_type(value_source, answers_with_context):
+def resolve_value_source_json_type(value_source, schema):
     source = value_source["source"]
     if source == "answers":
         answer_id = value_source["identifier"]
-        answer_type = answers_with_context[answer_id]["answer"]["type"]
+        answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
         return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
 
+    if source == "calculated_summary":
+        answer_type_list = []
+        block_id = schema.get_block(value_source["identifier"])
+        answer_ids = block_id["calculation"]["answers_to_calculate"]
+        for answer_id in answer_ids:
+            answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
+            answer_type_list.append(ANSWER_TYPE_TO_JSON_TYPE[answer_type])
```

Comment:
We only need to check 1 answer id. We should have other validation to ensure answers within calculated summary are of all same type (existing gap, so separate card).

Reply:
Updated this 👍

Reply:
to avoid using `0` index of the list we can potentially do something like:
```python   if source == "calculated_summary":
        block_id = schema.get_block(value_source["identifier"])
        for answer in block_id["calculation"]["answers_to_calculate"].keys():
            answer_id = block_id["calculation"]["answers_to_calculate"][answer]
            answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
            return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
```
so you only go through the loop once (and return early).

Reply:
Updated to use `next(iter(block_id["calculation"]["answers_to_calculate"]))` as discussed 👍